### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by rugwizangoga**
# React capstone project - countries-basic-info-app

- Created homepage when it loads, countries-basic-info-app shows the list of official languages filtered out by countries
- Created details page that lists those languages and other basic information about that country
- Added tests for both pages
- Documented the app with a clear README file and video
- Create reducer and store using react redux toolkit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,204 @@
...
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 [countries-basic-info-app] <a name="about-project"></a>
+
+**[countries-basic-info-app]** is a simple mobile react SPA that displays all countries around the world and the number official languages in those countries, it also provides the basic information about them like their capitals, location, currencies and so on.
+
+- Here is the introduction [video](https://www.loom.com/share/d2b596bc020341529f0a00490da22f68) about feature of the App.
```
Comment (Ghiftee) [inline README.md:57]: - [ ] Good job creating a video with all the required features, however, your video is currently longer than 5 minutes which is the required length. Kindly shorten the video without removing all the other required features.

File: src/App.css [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  width: 100%;
+  max-width: 100%;
+  max-height: 550px;
+  min-height: 300px;
+}
+
+.whole {
+  min-width: 450px;
+  height: 100vh;
+  background-color: #d04379;
```
Comment (Ghiftee) [inline src/App.css]: - [ ] According to the requirements, your app should be a mobile application and not desktop as it currently is. Kindly fix this

### Project requirements
Screenshot 2023-06-03 at 21 30 52 (2)


### Your project on mobile
Screenshot 2023-06-03 at 21 29 12 (2)

File: src/components/pages/Countries.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    dispatch(fetchCountries());
+  }, [dispatch]);
+
+  const sortedCountriesList = [...countriesList]
+    .sort((a, b) => a.name.common.localeCompare(b.name.common));
+
+  return (
+    <div className="app">
+      <img src={Mymap} alt="World map" className="wmap" />
+      <h1>Number of official languages per country</h1>
```
Comment (Ghiftee) [inline src/components/pages/Countries.js]: - [ ] You are currently missing this requirement - `one page with a list of items that could be filtered by some parameters; in the example below, it's a list of metrics that can be filtered by the country (imagine a search field to introduce the country name like Italy, Croatia, etc.). This page should be your homepage.` Please add it as it improves the ux of your application.

File: src/components/pages/Countries.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// Countries.js
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchCountries } from '../../redux/Countries/countriesSlice';
+import Mymap from '../../assets/wmap.png';
+import Icon from '../../assets/icon';
+
+const Countries = () => {
```
Comment (Ghiftee) [inline src/components/pages/Countries.js:8]: - [ ] Kindly do not forget to implement this requirement - `You should filter the data that you retrieve from the API using a Filter stateless component.`

File: src/App.css [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+.details {
+  align-self: center;
+}
+
+.wmap {
+  width: 100%;
+  max-width: 100%;
+  max-height: 550px;
+  min-height: 300px;
```
Comment (Ghiftee) [inline src/App.css]: - [ ] Kindly ensure that the size of the map section is smaller and follows the size of the design.

### Your project
Screenshot 2023-06-03 at 21 29 12 (2)


### The design
Screenshot 2023-06-03 at 21 41 43 (2)

File: src/App.css [hunk 6]
```diff
@@ -0,0 +1,120 @@
+* {
+  text-decoration: none;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  font-family: Verdana, Geneva, Tahoma, sans-serif;
```
Comment (Ghiftee) [inline src/App.css:6]: - [ ] According tot he requirements, your app should match the given design in typography: font face, size, and weight; Layout: composition and space between elements. Kindly ensure your app complies with this.

### Recommended font family
Screenshot 2023-06-03 at 21 57 32 (2)

File: src/tests/Countries.test.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import store from '../redux/store'; // Replace with your actual Redux store import
+
+test('renders Country component correctly', () => {
+  const { container } = render(
+    <Provider store={store}>
+      <Countries />
+    </Provider>,
+  );
+  expect(container).toMatchSnapshot();
+});
```
Comment (Ghiftee) [inline src/tests/Countries.test.js:14]: - [ ] Kindly ensure that more tests particularly unit tests are created for several components of the app as they allow for verification of individual component of an application.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @rugwizangoga 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ App is deployed
✔️ App contains the required pages
✔️ README contains acknowledgement section

I know you have put in a lot of effort, well done! However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @rugwizangoga, 


In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- Design is in line with the given design :+1: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by rugwizangoga** [comment]
@FrederickMih thank you I appreciate the review

**Pull request merged**